src/components/PostDetails.tsx

```diff
@@ -1,118 +1,138 @@
-              >
-                delete button
-              </button>
-            </div>
+        <p data-cy="PostBody">{activePost?.body}</p>
+      </div>
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+      <div className="block">
+        {AreCommentsLoading && <Loader /> }
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a
-                href="mailto:misha@mate.academy"
-                data-cy="CommentAuthor"
-              >
-                Misha Hrynko
-              </a>
+        {error && (
+          <div
+            className="notification is-danger"
+            data-cy="CommentsError"
+          >
+            {error}
 
+            {error !== ServerErrors.Comments && (
               <button
-                data-cy="CommentDelete"
                 type="button"
-                className="delete is-small"
-                aria-label="delete"
+                className="button button-retry"
+                onClick={onRetry}
               >
-                delete button
+                Try again
               </button>
-            </div>
-            <div
-              className="message-body"
-              data-cy="CommentBody"
+            )}
+
+            <button
+              type="button"
+              className="delete is-small"
+              aria-label="delete"
+              onClick={() => setError(null)}
             >
-              One more comment
-            </div>
-          </article>
+              delete button
+            </button>
+          </div>
+        )}
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a
-                href="mailto:misha@mate.academy"
-                data-cy="CommentAuthor"
-              >
-                Misha Hrynko
-              </a>
+        {isRetryLoading && <Loader />}
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+        {successfulLoad && (
+          !comments.length ? (
+            <p className="title is-4" data-cy="NoCommentsMessage">
+              No comments yet
+            </p>
+          ) : (
+            <>
+              <p className="title is-4">Comments:</p>
 
-            <div className="message-body" data-cy="CommentBody">
-              {'Multi\nline\ncomment'}
-            </div>
-          </article>
+              {comments.map(commentator => {
+                return (
+                  <article
+                    className="message is-small"
+                    data-cy="Comment"
+                    key={commentator.id}
```

Comment:
use destructuring

Reply:
Since some of the commentator properties are destructured above it was not done. It takes me to create a new destructuring with key-value pairs, so idk what is better, to keep this new destructed version or leave the primary version

Reply:
oh, I missed that it's `map`. so, the previous version was okay. and another option is that you can destructure in another way, just instead of `comments.map(commentator` you could try `comments.map({id, email, and so on}`